CHANGELOG.md

```diff
@@ -1,3 +1,31 @@
+### pg_auto_failover v1.4.0 (September 2, 2020) ###
+
+* Make pg_autoctl the parent process for Postgres [#265, #267, #284]
+* Allow adding a standby node from an exiting PGDATA directory [#276]
+* Improve output for many pg_autoctl commands, including JSON output
+* Fix pg_autoctl perform failover/switchover default --group [#289]
+* Fix pgautofailover extension upgrade script from 1.2 to 1.3 [#288]
+* Command pg_autoctl perform failover now waits until failover is done [#290]
+* Improve connection retry attempts [#299, #302]
+* Skip network DNS probes on pg_autoctl create monitor --skip-pg-hba [#298]
+* Fix pg_autoctl perform failover [#294, #307]
+* Implement pg_autoctl enable maintenance --allow-failover [#306]
+* Do not always require --ssl-ca-file for custom SSL [#303]
+* Review the registering process & transaction [#309]
+* Edit the HBA file as soon as a new node is added to the group [#311]
+* Check for monitor extension version every time we connect [#318]
+* Fix usage of replication slots to avoid a Postgres bug [#321]
+* Log and notify node health changes [#334]
+* Rename --nodename to --hostname, and nodename to nodehost [#273]
+* Implement a node name [#340]
+* Test primary_conninfo before starting Postgres standby [#350]
+* Allow multiple standbys in a Postgres group [#326]
```

Comment:
I would move this to the top, since it seems the most important feature. Given the amount of changes, I think it might even be nice to split the changelog section in multiple subsections like shown in the last bullet point here:
https://keepachangelog.com/en/0.3.0/#what-makes-a-good-change-log

Reply:
I have redone the CHANGELOG entries with following the model you're pointing to. I have also added a full section of text with some background information about the main changes.

One thing that we have not covered anywhere yet is the `sysidentifier`. That's a new column, and all nodes in a given group should have the same value there. At upgrade, the script sets it to zero, because we don't know the actual value from the nodes. In order to add a third node in a group after an upgrade, it will be necessary to set the `sysidentifier` column correctly on all nodes from a group at once. 

I'm not sure how to best fix that. Minimum amount of support would be to add an entry in the FAQ and/or the upgrade notes and show the needed SQL query to run on the monitor. Other than that, we could include a “fix system identifier at startup” feature, I guess, and relax our constraint to accept zero in all cases.

Reply:
So I have now added a “fix system identifier at startup” feature, that only kicks-in when the current value is zero and then change all the nodes to the same value (the local one). For this to work the exclusion constraint that protects this column is now “deferrable initially deferred”.